### DISCUSSION TITLE
what is the range SMuRF tracking can handle

### DISCUSSION THREAD

**Discussion by yuhanwyhan** [simonsobs/sodetlib#348] [Issues and Debugging]
what is the range SMuRF tracking can handle?
for example, if resonators shift 4KHz (or some other frequency), will SMuRF still be able to use the old tunefile and track?
Related to this, what is the upper bar of resonator frequency shift such that a tunefile should be created?

**Comment by jlashner** [comment]
I can't seem to find the exact plots right now.... but I remember increasing UC atten can change the resonance frequency by ~~kHz~~ ~100kHz, and that definitely requires a new tunefile, so I think that may be an upper bound. @msilvafe do you have plots from when we were testing that? Hopefully @swh76 has a better idea for what the spec on that is like.

**Comment by swh76** [comment]
Hard to say, best to do some focused testing.  If you rely on the serial gradient descent method to refine the resonator frequency estimate, it'll be determined by how well that routine is able to do.  That makes it tricky to say _a priori_ how well it will do for a given frequency shift because it's an iterative algorithm whose performance may be strongly impacted by measurement noise, resonator parameters, and the algorithm's parameters, which if you haven't, you may need to optimize.

If the shifts you're worried about are less than ~100 kHz, an easy way to test this might be to run a bunch of tunes vs fixed flux ramp bias (which will move all of the resonators in frequency in a controllable way) and see how well your tuning procedure does.

**Comment by yuhanwyhan** [comment]
what should be guiding the resonator re-tracking and setting up updated tunes? What should be boundary between only running retracking and running new setup notches?

**Reply by jlashner** [reply]
I think this will probably take some testing at site to figure out. If tunes are stable like they were for MF1 testing, then after attens are optimized I don't think we'll need to regularly re-run setup notches at all unless something with the cabling or setup is changed. 

If we're seeing frequency drifts with temperature that are too large to recover from with serial gradient descent, then maybe we want to build retunes into the schedule.

But I think we can just use noise levels over time to determine whether we need a new tune...

**Reply by yuhanwyhan** [reply]
I agree but I do think a number for reference will help this process.

**Reply by jlashner** [reply]
Yeah..... I'm pretty sure we just don't have a number unfortunately
